system/Helpers/array_helper.php

```diff
@@ -46,9 +46,10 @@ function dot_array_search(string $index, array $array)
     function _array_search_dot(array $indexes, array $array)
     {
         // Grab the current index
-        $currentIndex = $indexes ? array_shift($indexes) : null;
+        $currentIndex    = $indexes ? array_shift($indexes) : null;
+        $intCurrentIndex = (int) $currentIndex;
 
-        if ((empty($currentIndex) && (int) $currentIndex !== 0) || (! isset($array[$currentIndex]) && $currentIndex !== '*')) {
+        if (empty($currentIndex) && $intCurrentIndex !== 0 || (! isset($array[$currentIndex]) && $currentIndex !== '*')) {
```

Comment:
If `$currentIndex` is 0, `empty($currentIndex)` is true and `$intCurrentIndex !== 0` is false.
If `$currentIndex` is not 0, `empty($currentIndex)` is false and `$intCurrentIndex !== 0` is true.
So the left side is still always false.

Reply:
Does this seem better?
```php
        // Grab the current index
        $currentIndex = $indexes ? array_shift($indexes) : null;

        if ($currentIndex === null || (! isset($array[$currentIndex]) && $currentIndex !== '*')) {
            return null;
        }
```

Reply:
I'm sorry, I didn't realize it changed between my review and the merge. This was originally just removing the extraneous part of the conditional. 

Yours is better but this still has a lot of cognitive complexity and anyone working on it in the future will need to spend some time figuring it out. Maybe we should break it up into separate return statements? Or extract the second half of the conditional as a variable with a helpful name?

Reply:
Okay, I will fix this.

Reply:
I sent a PR #6645